src/catchup/simulation/ApplyTransactionsWork.cpp

```diff
@@ -63,14 +64,125 @@ checkResults(Application& app, uint32_t ledger,
         auto const& res = results[i];
         assert(res.transactionHash == resSet.results[i].transactionHash);
 
-        auto const& dbRes = resSet.results[i].result;
-        if (dbRes.result.code() != res.result.result.code())
+        auto const& dbRes = resSet.results[i].result.result;
+        if (dbRes.code() != res.result.result.code())
         {
             auto msg = fmt::format(
                 "Expected result code {} does not agree with {} for tx {}",
-                res.result.result.code(), dbRes.result.code(),
+                res.result.result.code(), dbRes.code(),
                 binToHex(res.transactionHash));
             CLOG(ERROR, "History") << msg;
+            return;
+        }
+
+        // Now check each operation result
+        auto const& archiveRes = results[i].result.result;
+
+        if (dbRes.code() == txFAILED || dbRes.code() == txSUCCESS)
+        {
+            assert(archiveRes.code() == txFAILED ||
```

Comment:
I don't think you can always guarantee this. It's possible, for example, that a transaction that changes signers succeeds in production but fails in simulation, then causes another transaction that failed in production with `txFAILED` to fail in simulation with `txBAD_AUTH`.

Reply:
it's already handled on line 68 (case where the codes aren't equal). the assert is redundant here, so i removed it.